src/main/scala/com/cognite/sdk/scala/common/dataTypes.scala

```diff
@@ -23,24 +23,30 @@ final case class SdkException(
     uri: Option[Uri] = None,
     requestId: Option[String] = None,
     responseCode: Option[Int] = None
-) extends Throwable(message) {
-  override def toString: String = {
+) extends Throwable(SdkException.formatMessage(message, uri, requestId, responseCode))
+object SdkException {
+  def formatMessage(
+      message: String,
+      uri: Option[Uri],
+      requestId: Option[String],
+      responseCode: Option[Int]
+  ): String = {
     val responseCodeMessage = responseCode
       .map(c => s" (with response code ${c.toString})")
       .getOrElse("")
     val uriMessage = uri
       .map(u => s", in response$responseCodeMessage to request sent to ${u.toString()}")
       .getOrElse(responseCodeMessage)
     val requestIdMessage = requestId.map(id => s", with request id $id").getOrElse("")
-    val superString = super.toString
-    val superMessage = if (superString.length > 0) {
-      superString
+    val superMessage = if (message.length > 0) {
```

Comment:
This no longer has anything to do with super, so call it `exceptionMessage` (or something like that), or rename the parameter to `originalMessage: String` and call this one `val message = if (originalMessage.length > 0)...`